pkg/asset/installconfig/vsphere/client.go

```diff
@@ -39,31 +39,37 @@ func NewFinder(client *vim25.Client, all ...bool) Finder {
 	return find.NewFinder(client, all...)
 }
 
+// ClientLogout is empty function that logs out of vSphere clients
+type ClientLogout func()
+
 // CreateVSphereClients creates the SOAP and REST client to access
 // different portions of the vSphere API
 // e.g. tags are only available in REST
-func CreateVSphereClients(ctx context.Context, vcenter, username, password string) (*vim25.Client, *rest.Client, error) {
+func CreateVSphereClients(ctx context.Context, vcenter, username, password string) (*vim25.Client, *rest.Client, ClientLogout, error) {
 	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
 	defer cancel()
 
 	u, err := soap.ParseURL(vcenter)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	u.User = url.UserPassword(username, password)
-	c, err := govmomi.NewClient(ctx, u, false)
+	client, err := govmomi.NewClient(ctx, u, false)
 
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	restClient := rest.NewClient(c.Client)
+	restClient := rest.NewClient(client.Client)
 	err = restClient.Login(ctx, u.User)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
```

Comment:
If there is an error creating the rest client, the govmomi client logout will not happen.